.github/workflows/lint.yaml

```diff
@@ -13,22 +13,46 @@
 # limitations under the License.
 
 name: lint
-on: [pull_request]
+on:
+  pull_request:
+  pull_request_target:
+    types: [labeled]
 
 jobs:
   build:
-    name: lint
+    if: "${{ github.event.action != 'labeled' || github.event.label.name == 'tests: run' }}"
+    name: Run lint
     runs-on: ubuntu-latest
     steps:
+      - name: Remove PR Label
+        if: "${{ github.event.action == 'labeled' && github.event.label.name == 'tests: run' }}"
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            try {
+              await github.rest.issues.removeLabel({
+                name: 'tests: run',
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: context.payload.pull_request.number
+              });
+            } catch (e) {
+              console.log('Failed to remove label. Another job may have already removed it!');
+            }
       - name: Setup Go
         uses: actions/setup-go@v3
         with:
           go-version: "1.19"
-      - name: Install goimports
-        run: go install golang.org/x/tools/cmd/goimports@latest
       - name: Checkout code
         uses: actions/checkout@v3
-      - run: goimports -w .
-      - run: go mod tidy
-      - name: Verify no changes from goimports and go mod tidy. If you're reading this and the check has failed, run `goimports -w . && go mod tidy`.
-        run: git diff --exit-code
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: go mod tidy
```

Comment:
I don't like surprising developers with CI lint errors that can only be reproduced locally if you understand how to read the CI code. 

I prefer that there is single command that the local developer can run that will do an identical lint check to the CI. For operator, the Makefile was the most sensible way to do this. For other repos that aren't using make, perhaps a shell script would do.  

Would you consider adding a shell script that can run these same commands and add instructions in the README.md?

Reply:
We probably should add these commands to the CONTRIBUTING.md file.

Since they're all captured in CI, though, and I'm the only contributor so far, I'd like to postpone those changes. But otherwise, I agree. It should be easy to run all checks locally.